notes.md

```diff
@@ -191,13 +191,13 @@ Some general notes for a program:
 
 ## Default flag
 
-The "default track" flag is a hint for the playback application and SHOULD always be changeable by the user. If the user wants to see or hear a track of a certain kind (audio, video, subtitles) and she hasn't chosen a specific track then the player SHOULD use the first track of that kind whose "default track" flag is set to "1". If no such track is found then the first track of this kind SHOULD be chosen.
+The "default track" flag is a hint for the playback application and SHOULD always be changeable by the user. If the user wants to see or hear a track of a certain kind (audio, video, subtitles) and she hasn't chosen a specific track (either explicitly, or via a language preference), then the player SHOULD use the first track of that kind whose "default track" flag is set to "1". If no such track is found then the first track of this kind SHOULD be chosen.
 
 Only one track of a kind MAY have its "default track" flag set in a segment. If a track entry does not contain the "default track" flag element then its default value "1" is to be used.
 
 ## Forced flag
 
-The "forced" flag tells the playback application that it MUST display/play this track or another track of the same kind that also has its "forced" flag set. When there are multiple "forced" tracks, the player SHOULD determined based upon the language of the forced flag or use the default flag if no track matches the use languages. Another track of the same kind without the "forced" flag may be use simultaneously with the "forced" track (like DVD subtitles for example).
+If a track with the "forced" flag whose language matches the user's preference exists, and the user's preferences would normally indicate that subtitles would not be displayed with the selected audio track, the "forced" track SHOULD be displayed by default.
```

Comment:
A forced flag is really forced. It can't be overridden. Only a different flavor of a forced flag could replace it. And if there are many forced flags, then the one with the default flag takes precedence.

Reply:
Ping on this; I'd like to get this PR updated, but I'm not sure how to address this comment. No real players treat "forced" as not being user-overridable (including with a non-forced track), and I don't think there's any appetite to change that.

Reply:
Sorry for the huge delay.
"forced" tracks are user hostile, yes. It can happen on some DVDs where they force you to see a certain subtitle. I also noticed on iTunes some movies have forced subtitles, you can't turn them off.

It can also be nice in some cases. For example if in the movie a person talks an alien language that will be shown as a subtitle even though you don't have subtitles one. But that raises another subject. In what language should it be displayed. In that case there could be multiple variants of the same (klingon) subtitle track. Each of them should have the forced flag. That's where this section comes from to pick the best one.

Reply:
> It can happen on some DVDs where they force you to see a certain subtitle.

Sure, but Matroska isn't DVD and I don't think there's any value in trying to imitate it here.

> I also noticed on iTunes some movies have forced subtitles, you can't turn them off.

This is probably implemented using the "forced" flag in HLS, which is also somewhat poorly-defined; even within Apple there are multiple different implementations with different behaviors (iTunes vs Safari vs QuickTime Player). But there's nothing stopping an end-to-end system like iTunes from preventing the user from disabling forced tracks, even if the spec didn't explicitly require it.

> It can also be nice in some cases. For example if in the movie a person talks an alien language that will be shown as a subtitle even though you don't have subtitles one. 

Right, and that's how the flag is used in practice (when it's used for anything meaningful at all, anyway).

> In what language should it be displayed. In that case there could be multiple variants of the same (klingon) subtitle track. Each of them should have the forced flag.

Yeah, that's what I'm trying to specify; the appropriate forced track for the user's selected language should be selected _by default_ by the player, unless the user's subtitle preferences cause a non-forced track to be used instead. I take issue with (a) making it a requirement that they can't be turned off, which no player implements and I expect no mainstream player ever will, and (b) specifying that multiple tracks should be superimposed, which no player implements and no player _should_ implement (since it would break essentially all existing files that use forced streams in this way).

I'd really like the spec to actually reflect reality here, and this "forced means literally forced" thing doesn't. A "MUST" in the spec is meaningless if every implementation decides to ignore it, and I can tell you that mpv and Plex certainly aren't going to implement that behavior. Along with how it'd result in unhappy users for no benefit, it would break a decent number of files that already exist in the wild and use the "forced" flag on regular subtitle streams.